### PR TITLE
fix(share_plus): Fix 'share text' not showing when share files

### PR DESCRIPTION
**Pull request by salahamassi**
## Description

when the user uses `shareFiles` or  `shareWithResult`  methods to share a file or files,
`text` argument was ignored and not added to `share:items` on the IOS platform, so I just add the text to `share:items` .

## Related Issues

*no related issues

## Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [X] All existing and new tests are passing.
- [X] I updated the version in `pubspec.yaml` and `CHANGELOG.md`.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [X] I read and followed the [Flutter Style Guide].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

### CODE REVIEW

**Review by rafampessoa — APPROVED** [review]

File: packages/share_plus/share_plus/ios/Classes/FLTSharePlusPlugin.m [hunk 1]
```diff
@@ -368,6 +370,13 @@ + (void)shareFiles:(NSArray *)paths
                                                 mimeType:mimeType
                                                  subject:subject]];
   }
+  if (text != nil) {
+    NSObject *data = [[NSURL alloc] initWithString:text];
+    if (data == nil) {
```
Comment (miquelbeltran) [inline packages/share_plus/share_plus/ios/Classes/FLTSharePlusPlugin.m:375]: Sorry, I don't understand this code block. `data` is never going to be `nil`, right?

Comment (salahamassi) [reply packages/share_plus/share_plus/ios/Classes/FLTSharePlusPlugin.m:375]: no data can be nill if the text is not valid NSURL
in this case (text is not valid NSURL) we create a SharePlusData with the given text

Comment (miquelbeltran) [reply packages/share_plus/share_plus/ios/Classes/FLTSharePlusPlugin.m:375]: got it, thanks!

**Comment by miquelbeltran** [comment]
I've been testing this, and I have no problem approving and merging, but I fail to see what problem is it solving.

When I test this PR example against multiple apps: Telegram, Google Keep, Twitter, etc. sharing both images and text, only the image gets shared. 

I am unable to see if this is a limitation with share_plus or a general problem with all apps, and I am not sure if this PR attempts to fix this problem or something else.

**Comment by salahamassi** [comment]
> I've been testing this, and I have no problem approving and merging, but I fail to see what problem is it solving.
> 
> When I test this PR example against multiple apps: Telegram, Google Keep, Twitter, etc. sharing both images and text, only the image gets shared.
> 
> I am unable to see if this is a limitation with share_plus or a general problem with all apps, and I am not sure if this PR attempts to fix this problem or something else.


I attached two videos to show the difference before and after this PR when I try to share a text with an image through the Email or Twitter apps.

These videos are captured from my iPhone 12 ios 15.5

**before video**
https://user-images.githubusercontent.com/17902030/175652435-d37591d4-c9e0-4d40-8b79-c21a5eef6d58.mov

**after video**
https://user-images.githubusercontent.com/17902030/175652931-bff71437-1c00-44de-bf07-6b17e741e8ff.mov

The code added to this PR code adds the share text to `share:items` array.

some apps like Twitter, email, or telegram app will display the share text with an image, and some apps will not read the text just the image like the Facebook app (the Facebook app will read just hashtag text like #salah I don't know why 😐),  and WhatsApp app will just read the text and will ignore the image.

I believe each app will handle the share items in it's custom way.

but of course, we need to add the share text to  `share:items` array so the apps can know there is the text along with the shared image.

**Comment by miquelbeltran** [comment]
Thanks for the videos! Something was wrong when I tested this then, because I couldn't see the shared text on Twitter, but I see that it works on your case

**Comment by miquelbeltran** [comment]
We forgot to update the pubspec.yml before merging, I will fix that on main directly

**Comment by miquelbeltran** [comment]
4.0.10 published now

**Pull request merged**